src/main/scala/io/iohk/ethereum/blockchain/data/GenesisDataLoader.scala

```diff
@@ -115,19 +115,57 @@ class GenesisDataLoader(
         Failure(new RuntimeException("Genesis data present in the database does not match genesis block from file." +
           " Use different directory for running private blockchains."))
       case None =>
-        ephemDataSource.getAll(nodeStorage.namespace)
-          .foreach { case (key, value) => blockchain.saveNode(ByteString(key.toArray[Byte]), value.toArray[Byte], 0) }
         blockchain.save(Block(signedHeader, BlockBody(Nil, Nil)), Nil, signedHeader.header.difficulty, saveAsBestBlock = true)
         Success(())
     }
   }
 
-  private def prepareSignedHeader(genesisData: GenesisData, stateMptRootHash: Array[Byte]) = {
-    val genesisHeader = BlockHeader(
+  private def storeUserAllocAccounts(nodeStorage: NodeStorage, genesisData: GenesisData, initialRootHash: Array[Byte]): Array[Byte] = {
+    import MerklePatriciaTrie.defaultByteArraySerializable
+    genesisData.alloc.zipWithIndex.foldLeft(initialRootHash) {
+      case (rootHash, (((address, AllocAccount(balance)), idx))) =>
+        val ephemNodeStorage = pruning.PruningMode.nodesKeyValueStorage(pruning.ArchivePruning, nodeStorage)(Some(idx - genesisData.alloc.size))
+        val mpt = MerklePatriciaTrie[Array[Byte], Account](rootHash, ephemNodeStorage)
+        val paddedAddress = address.reverse.padTo(addressLength, "0").reverse.mkString
+        mpt.put(crypto.kec256(Hex.decode(paddedAddress)),
+        Account(blockchainConfig.accountStartNonce, UInt256(BigInt(balance)), emptyTrieRootHash, emptyEvmHash)
+      ).getRootHash
+    }
+  }
+
+  private def deployConsensusContract(consensusContractData: ContractData, preBlockHeader: BlockHeader, currentStateRootHash: ByteString): ByteString = {
+    val stx = createSignedContractTransaction(consensusContractData.code)
+    val evmConfig = EvmConfig.forBlock(blockNumber = BigInt(0), blockchainConfig)
+    val worldWithContractAccount = blockchain
+      .getWorldStateProxy(BigInt(0), blockchainConfig.accountStartNonce, Some(currentStateRootHash))
+      .saveAccount(consensusContractData.address, Account())
+    val context: PC = ProgramContext(stx, consensusContractData.address, Program(stx.tx.payload), preBlockHeader, worldWithContractAccount, evmConfig)
+    val result = virtualMachine.run(context)
+    log.debug(s"Error while deploy consensus contract: ${result.error.toString}")
```

Comment:
Minor detail: Maybe we should change this to `error` level and only print it if there was any error during execution?